rclc_parameter/include/rclc_parameter/rclc_parameter.h

```diff
@@ -365,6 +452,109 @@ rclc_parameter_get_double(
   const char * parameter_name,
   double * output);
 
+
+/**
+ * Add a description to a parameter (Unsuported on low memory mode).
+ * This description will be returned on describe parameters requests.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * \param[in] parameter_server preallocated rclc_parameter_server_t
+ * \param[in] parameter_name name of the parameter
+ * \param[in] parameter_description description of the parameter
+ * \param[in] additional_constraints constraints description
+ * \param[in] read_only read only flag
+ * \return `RCL_RET_OK` if success
+ */
+RCLC_PARAMETER_PUBLIC
+rcl_ret_t
+rclc_add_parameter_description(
+  rclc_parameter_server_t * parameter_server,
+  const char * parameter_name, const char * parameter_description,
+  const char * additional_constraints);
+
+/**
+ * Sets a parameter read only flag.
+ * Read only parameters can only be modified from the `rclc_parameter_set` API.
+ * This method is disabled on user callback execution.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * \param[in] parameter_server preallocated rclc_parameter_server_t
+ * \param[in] parameter_name name of the parameter
+ * \param[in] read_only read only flag
+ * \return `RCL_RET_OK` if success
+ */
+RCLC_PARAMETER_PUBLIC
+rcl_ret_t
+rclc_set_parameter_read_only(
+  rclc_parameter_server_t * parameter_server,
+  const char * parameter_name, bool read_only);
+
+/**
+ * Add constraint description for an integer parameter.
+ * This constraint description will be returned on describe parameters requests.
+ * This method is disabled on user callback execution.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * \param[in] parameter_server preallocated rclc_parameter_server_t
+ * \param[in] parameter_name name of the parameter
+ * \param[in] from_value start value for valid values, inclusive.
+ * \param[in] to_value end value for valid values, inclusive.
+ * \param[in] step size of valid steps between the from and to bound.
+ * \return `RCL_RET_OK` if success
+ */
+RCLC_PARAMETER_PUBLIC
+rcl_ret_t
+rclc_add_parameter_constraints_integer(
```

Comment:
```suggestion
rclc_add_parameter_constraint_integer(
```
I would call it one constraint only (i.e., singular), even it covers minimum, maximum, and step.

Reply:
Fine by me: 50c4a127f19375e24892fd611547dd5ba5195e9a